checker/check_result.go

```diff
@@ -97,6 +110,42 @@ func CreateProportionalScore(success, total int) int {
 	return int(math.Min(float64(MaxResultScore*success/total), float64(MaxResultScore)))
 }
 
+// CreateProportionalScoreWeighted creates the proportional score
+// between multiple successes over the total, but some proportions
+// are worth more.
+func CreateProportionalScoreWeighted(scores ...ProportionalScoreWeighted) (int, error) {
+	var ws, wt int
+	allWeightsZero := true
+	noScoreGroups := true
+	for _, score := range scores {
+		if score.Success > score.Total {
+			return InconclusiveResultScore, fmt.Errorf("%w: %d, %d", ErrSuccessTotal, score.Success, score.Total)
+		}
+		if score.Total != 0 {
+			noScoreGroups = false
+		} else {
+			// Group with 0 total, does not count for score
+			continue
+		}
```

Comment:
this is super nit and optional, but in general [keep the happy path left](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88).

```go
if score.Total == 0 {
    continue // Group with 0 total, does not count for score
}
noScoreGroups = false
```